### PR TITLE
[Merged by Bors] - p2p: fix connections reporter

### LINKED ISSUE #4225: GRPC NodeService.StatusStream does not send updates on ConnectedPeers anymore
## Description
Currently `NodeService.StatusStream` sends updates only on progression of layers and does not notify about changes in the amount of connected peers, as it was before. I assume the regression was introduced within #4108 

## Steps to reproduce
1. Run go-spacemesh v0.2.25-beta.0 against some working network
2. Run `grpcurl --plaintext localhost:9092 spacemesh.v1.NodeService.StatusStream`

## Actual Behavior
It sends updates only on layer changes (e.g. `topLayer` increments).

## Expected Behavior
It should send updates also for changing connected peers amount, as it was before.

## Environment
 - Node Version: 0.2.25-beta.0

## Additional Resources 
```
$ grpcurl --plaintext localhost:9092 spacemesh.v1.NodeService.StatusStream
{
  "status": {
    "connectedPeers": "5",
    "syncedLayer": {
      "number": 239
    },
    "topLayer": {
      "number": 10
    },
    "verifiedLayer": {
      "number": 239
    }
  }
}
{
  "status": {
    "connectedPeers": "10",
    "syncedLayer": {
      "number": 239
    },
    "topLayer": {
      "number": 11
    },
    "verifiedLayer": {
      "number": 239
    }
  }
}
```

p.s.
I didn't check for updates on `syncedLayer` and `verifiedLayer` yet, but hopefully, it works :D

### PR DESCRIPTION
**Pull request by dshulyak**
closes: https://github.com/spacemeshos/go-spacemesh/issues/4225

### CODE REVIEW

File: p2p/upgrade_test.go [hunk 1]
```diff
@@ -46,3 +50,32 @@ func TestIsBootnode(t *testing.T) {
...
+	reporter := func(i int) func() {
+		return func() {
+			counter[i].Add(1)
+		}
+	}
+
+	for i, host := range mesh.Hosts() {
+		_, err := Upgrade(host, types.Hash20{}, WithNodeReporter(reporter(i)))
+		require.NoError(t, err)
+	}
```
Comment (poszu) [inline p2p/upgrade_test.go]: NIT - is the extra factory needed?
```suggestion
	for i, host := range mesh.Hosts() {
	       i := i
		_, err := Upgrade(host, types.Hash20{}, WithNodeReporter( func() { counter[i].Add(1) }))
		require.NoError(t, err)
	}
```

**Review by poszu — APPROVED** [review]

File: p2p/upgrade_test.go (hunk 1 above)
Comment (dshulyak) [reply p2p/upgrade_test.go]: thanks, makes sense

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**